### ISSUE TITLE
Some interface for limiting actor's queue size is required

### ISSUE THREAD

**Issue by svetlyak40wt** [mdbergmann/cl-gserver#87]
Here is my suggestion:

```lisp
(setf *test-actor*
        (actor-of *system*
                  :receive #'run-test-func
                  :init #'init-test-func
                  :destroy #'destroy-test-func
                  :state :foo
                  :name "test-actor"
                  :other-args '(:queue-size 1))))
```

Here we pass queue-size in `other-args`.

And then use it in the `actor-of` method:

```
(defmethod actor-of ((context actor-context)
                     &key receive
                          (init nil) (destroy nil)
                          (dispatcher :shared) (state nil)
                          (type 'act:actor) (name nil)
                          (other-args nil))
  "See `ac:actor-of`."
  (check-type receive function "a function!")
  (%actor-of context
             (lambda () (act:make-actor receive
                                        :state state
                                        :name name
                                        :type type
                                        :init init
                                        :destroy destroy
                                        :other-init-args other-args))
             :dispatcher dispatcher
             :queue-size (getf other-args :queue-size)))
```

Probably it should be an explicit keyword argument of the `actor-of` generic function. What do you think, @mdbergmann ?

**Comment by mdbergmann** [comment]
Passing it in via `other-args` is OK for me. Maybe there are more 'other' args in the future that can also use this mechanism.
I think this parameter is not important enough to make it an explicit keyword. But should be documented.

**Comment by svetlyak40wt** [comment]
Actually, the first time I saw `other-args`, I thought these other args will be passed to the actor's constructor like this:

```lisp
(apply #'make-instance type
       :name name
       :state state
       :receive receive
       :init init
       :destroy destroy
       other-init-args)
```

This way it will complement `type` argument of the `actor-of` generic-function, because `other-args` list will be used to fill arguments of the user class provided as `type` argument.

**Comment by mdbergmann** [comment]
They are not only to allow arguments for a specialized actor 'type' but also for the standard type for example when the actor needs some initialization data beyond what the interface of `actor-of` or `act:make-actor` allows.
In some other project I use it for example to allow some special initialization in `act:pre-start` from data passed in via `other-init-args`.

So using this here for the queue-size is a vehicle to pass queue-size just from `actor-of` to `%actor-of`. The actor will then also be constructed with the queue-size parameter as part of `other-init-args` but I think that's OK even if it is unused later.

**Comment by mdbergmann** [comment]
Not sure if you are suggesting to change the interface?
Let me think about it some more.

**Comment by svetlyak40wt** [comment]
I'm just saying that right now there is no other way to pass additional initialization arguments to a custom actor class other then to put them into `other-args` and then to put into actual slots in the `initialize-instance` method specialized on a custom class.

**Comment by svetlyak40wt** [comment]
But this is definitely a design choice. Why `other-args` slot are need at all if the data slots can be filled as real slots of a custom class? In case, if user does not want to create a custom class, but want to pass some values into an actor, then there is an initial state.

**Comment by mdbergmann** [comment]
I use also slots to store stuff in `other-init-args` in another actor type but have to pull them out of `other-init-args` via `getf`.
https://github.com/mdbergmann/chipi/blob/d74b88da9ad161615e495d51a209a2a59db3df4d/src/persistence-influx.lisp#L50

So in your way one could define `initialize-instance` explicitly with the named slots.

Yeah, I think I'd like that. The current solution was born out of a need to have custom parameters. But I didn't think about using `apply` to create the instance, which is kind of essential then.

**Comment by mdbergmann** [comment]
Well, I suppose I know what you mean, when slots are defined with `:initarg` and you supply additional keyword list via `make-instance` and `apply` they are copies directly to the slots.
Sounds good. Looks like the current approach is just a workaround for this.

**Comment by svetlyak40wt** [comment]
Indeed! This exactly what I mean!

**Comment by svetlyak40wt** [comment]
Currently I'm buzy participating in a hackathon, but will make a PR during the next week.

**Comment by svetlyak40wt** [comment]
I think we could keep other-args for a backward compatibility, or deprecate it an issue a warning about deletion after some period of time.

**Comment by svetlyak40wt** [comment]
Here is a PR where `:other-args` was replaced with ability to pass any unknown args to an actor's class:
https://github.com/mdbergmann/cl-gserver/pull/91

**Comment by mdbergmann** [comment]
Fixed as part of #91
